plugins/async-node/core/src/index.ts

```diff
@@ -127,7 +149,12 @@ export class AsyncNodePluginPlugin implements AsyncNodeViewPlugin {
       const newNode = resolvedNode || node;
       if (!resolvedNode && node?.type === NodeType.Async) {
         queueMicrotask(async () => {
-          const result = await this.basePlugin?.hooks.onAsyncNode.call(node);
+          const result = await this.basePlugin?.hooks.onAsyncNode.call(
+            node,
+            (result) => {
+              this.updateAsyncFunc(node, result, options, this.currentView);
+            },
+          );
           const parsedNode =
```

Comment:
Can we also update this part to use the `updateAsyncFunc` since it performs the same logic?

Reply:
done